### ISSUE TITLE
PSR12.Properties.ConstantVisibility should automatically detect the PHP version

### ISSUE THREAD

**Issue by razvanphp** [PHPCSStandards/PHP_CodeSniffer#393]
## Is your feature request related to a problem?
When running PSR12 against an (older) project, we see this warning message for projects that actually support PHP versions lower than 7.1.

```
FILE: /Volumes/Code/yii2/framework/grid/GridView.php
------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 3 WARNINGS AFFECTING 3 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------
 52 | WARNING | Visibility must be declared on all constants if your project supports PHP 7.1 or later (PSR12.Properties.ConstantVisibility.NotFound)
```

In my particular case, this is defined in composer.json:
```
    "require": {
        "php": ">=5.4.0",
```
The php version running phpcs is not relevant, as it might run just in CI.

## Describe the solution you'd like
It would be great to skip this check if the minimum version is below PHP 7.1, ideally checking this in composer.json. Also, I think this is an easy fix for `phpcbf`, but it's not yet implemented, it should be `public` as this is the default visibility if it is omitted.

## Additional context (optional)
Can I please get some guidance of how this should be implemented? Saw in another issue this, but not sure it's reliable enough for this use-case:
https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9b6260b4ff9410682fd34a34e68e377e60dc4961/src/Standards/Generic/Sniffs/PHP/DisallowAlternativePHPTagsSniff.php#L43-L52

- [x] I intend to create a pull request to implement this feature.

**Comment by jrfnl** [comment]
@razvanphp Thanks for opening this issue.

> It would be great to skip this check if the minimum version is below PHP 7.1, ideally checking this in composer.json. 

PHPCS can not and should not rely on a `composer.json` file being available. PHPCS can be installed via PHIVE, git clone, stand-alone PHAR, Composer etc, so any solution which presumes the existence of a `composer.json` file in a project (and presumes a project-based install), will not be accepted.

> Also, I think this is an easy fix for phpcbf, but it's not yet implemented, it should be public as this is the default visibility if it is omitted.

As spelled out in the [CONTRIBUTING guide](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md#contributing-with-code), PHPCS does not allow risky fixers for its native sniffs. The appropriate visibility is something which should be deliberately determined by a developer. This is not something a tool can do and making automated BC-compatible fixes just to fix the issues instead of making deliberate choices about the visibility is a risky fix, so any PR for this will not be accepted.

I'm pretty sure there are sniffs available in external standards which can make that fix for you, but it will not be made by the PHPCS native sniffs.

> Can I please get some guidance of how this should be implemented ?

The code snippet shown, works based on a [configured `php_version`](https://github.com/PHPCSStandards/PHP_CodeSniffer/wiki/Configuration-Options#setting-the-php-version) and defaults to the PHP version used to run PHPCS.
The PHPCS version to run PHPCS is - in most cases - not the same as the minimum supported PHP version, so defaulting to that PHP version would not be appropriate.

**Comment by razvanphp** [comment]
ok, thanks for the detailed explanation. So no other options? I can't see a viable way to treat this case otherwise. Maybe the sniff could receive a parameter as the minimum PHP version?

**Comment by razvanphp** [comment]
OK, for now I will just exclude it in the [project](https://github.com/yiisoft/yii2/pull/20121) `phpcs.xml`, but if you think about a clever solution, I'm all in to try it out. Thank you!
